### PR TITLE
feat: add x-gts-final and x-gts-abstract schema modifier support

### PR DESCRIPTION
**Pull request by aviator5**
- Define section 9.11 with full semantics for final (no inheritance) and abstract (no direct instances) modifiers
- Update OP#6 and OP#12 descriptions to reference modifier enforcement
- Add x-gts-abstract to event and capability base schemas, x-gts-final to vm_state schema
- Add OP#12 tests for final type derivation rejection, mid-chain final, and sibling unaffected
- Add OP#6 tests for abstract type instance rejection (well-known, anonymous, derived concrete)
- Add cross-cutting tests for mutual exclusion and combined scenarios

## Summary by CodeRabbit

* **New Features**
  * Added `x-gts-final` and `x-gts-abstract` schema-only modifiers: final blocks derivation; abstract forbids direct instances (instances must use a concrete rightmost derived type).

* **Behavior & Requirements**
  * OP#6/OP#12 updated: instances with rightmost abstract must fail; deriving from any final base must fail. Modifiers must be top-level boolean true, mutually exclusive; non-boolean or in-instance placement rejected. Registration-time validation guarded.

* **Documentation**
  * Added section detailing semantics, placement, examples, and trait interactions.

* **Examples**
  * Marked example schemas as abstract/final and updated descriptions.

* **Tests**
  * Extensive tests for registration, derivation, instance validation, trait interactions, and edge cases.

### CODE REVIEW

File: tests/test_op12_schema_vs_schema_validation.py [hunk 1]
```diff
@@ -3570,5 +3570,150 @@ def test_start(self):
...
+                },
+            },
+            "register derived from final base",
+        ),
+        _validate_schema(
+            "gts.x.test12.final.base.v1~x.test12._.derived.v1~",
+            False,
+            "validate derived from final base should fail",
+        ),
+    ]
```
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_op12_schema_vs_schema_validation.py:3568]: Won't fix — HttpRunner framework requires `teststeps` to be a `list`, not a tuple. All test classes across the project use lists consistently. Converting to tuples would break the framework contract.

File: tests/test_op6_schema_validation.py [hunk 2]
```diff
@@ -858,5 +858,195 @@ def test_start(self):
...
+            RunRequest("validate instance of abstract type should fail")
+            .post("/validate-instance")
+            .with_json({
+                "instance_id": "gts.x.test6.abstract.base.v1~x.test6._.my_item.v1",
+            })
+            .validate()
+            .assert_equal("status_code", 200)
+            .assert_equal("body.ok", False)
+        ),
+    ]
```
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_op6_schema_validation.py:918]: Won't fix — same reasoning: HttpRunner requires `teststeps` as a mutable `list`. This is a framework convention across all test files.

File: README.md [hunk 3]
```diff
@@ -1500,6 +1504,95 @@ Ensure generated schemas use GTS identifiers as `$id` for types and keep any `x-
...
+
+**Mutual exclusion:** A schema MUST NOT declare both `"x-gts-final": true` and `"x-gts-abstract": true`. This combination is semantically meaningless (a type that can be neither inherited from nor instantiated serves no purpose) and MUST be rejected during schema registration or validation.
+
+| Modifier combination | Inheritance allowed? | Direct instances allowed? |
+|---|---|---|
+| `x-gts-abstract: true` | Yes | No |
+| _(default / neither)_ | Yes | Yes |
+| `x-gts-final: true` | No | Yes |
+| Both `true` | **INVALID** — MUST be rejected |
+
```
[the remark this answers is not in the document]
Comment (aviator5) [reply README.md:1541]: Fixed — replaced underscore emphasis with asterisks and added the missing third column to the "Both true" row.

File: tests/test_op12_schema_vs_schema_validation.py [hunk 4]
```diff
@@ -3570,5 +3570,150 @@ def test_start(self):
...
+        _register_derived(
+            "gts://gts.x.test12.finalmid.base.v1~x.test12._.mid.v1~",
+            "gts://gts.x.test12.finalmid.base.v1~",
+            {
+                "type": "object",
+                "x-gts-final": True,
+                "properties": {
+                    "midField": {"type": "string"},
+                },
+            },
```
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_op12_schema_vs_schema_validation.py:3599]: Fixed — added `top_level` parameter to `_register_derived` in this file (matching `test_refimpl`) and moved `x-gts-final` out of overlay into `top_level={"x-gts-final": True}` for both mid-chain and sibling test cases.

File: tests/test_refimpl_x_gts_final_abstract.py [hunk 5]
```diff
@@ -0,0 +1,878 @@
...
+            "gts://gts.x.testfa.finaltrait.base.v1~",
+            {
+                "type": "object",
+                "x-gts-final": True,
+                "x-gts-traits": {
+                    "priority": 5,
+                },
+            },
+            "register final derived with traits resolved",
+        ),
```
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_refimpl_x_gts_final_abstract.py:775]: Fixed — moved `x-gts-final` from overlay dict to `top_level={"x-gts-final": True}` parameter.

File: tests/test_refimpl_x_gts_final_abstract.py [hunk 6]
```diff
@@ -0,0 +1,878 @@
...
+            .with_json({
+                "$$id": "gts://gts.x.testfa.finalbadval.base.v1~",
+                "$$schema": "http://json-schema.org/draft-07/schema#",
+                "type": "object",
+                "x-gts-final": "yes",
+                "properties": {"name": {"type": "string"}},
+            })
+            .validate()
+            .assert_equal("body.ok", False)
+        ),
```
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_refimpl_x_gts_final_abstract.py:244]: Fixed — added `.assert_equal("status_code", 200)` to all three registration-rejection steps (non-boolean final, non-boolean abstract, both modifiers). Matches the pattern used in other test files.

File: tests/test_op6_schema_validation.py [hunk 7]
```diff
@@ -858,5 +858,195 @@ def test_start(self):
...
+        Step(
+            RunRequest("validate instance of abstract type should fail")
+            .post("/validate-instance")
+            .with_json({
+                "instance_id": "gts.x.test6.abstract.base.v1~x.test6._.my_item.v1",
+            })
+            .validate()
+            .assert_equal("status_code", 200)
+            .assert_equal("body.ok", False)
+        ),
```
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_op6_schema_validation.py:917]: Won't fix — the `/validate-instance` response shape is not guaranteed to include `body.id`. No existing tests in the repo assert `body.id` on validate-instance responses (only on `/entities` registration). Adding this assertion would couple tests to an unspecified response field and risk false failures.

File: tests/test_refimpl_x_gts_final_abstract.py [hunk 8]
```diff
@@ -0,0 +1,881 @@
...
+    """Validate an instance via POST /validate-instance."""
+    step = (
+        RunRequest(label)
+        .post("/validate-instance")
+        .with_json({"instance_id": instance_id})
+        .validate()
+        .assert_equal("status_code", 200)
+        .assert_equal("body.ok", expect_ok)
+    )
+    return Step(step)
```
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_refimpl_x_gts_final_abstract.py]: Won't fix — same reasoning as above. No existing tests assert `body.id` on `/validate-instance` responses, and the response contract doesn't specify this field. Adding an optional `expected_id` parameter would add complexity for a speculative assertion.

File: tests/test_op6_schema_validation.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_op6_schema_validation.py:917]: You're right — existing tests in this file do assert `body.id` on `/validate-instance` failure paths. Fixed: added `body.id` assertions to both abstract-rejection validate-instance steps.

File: tests/test_refimpl_x_gts_final_abstract.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (aviator5) [reply tests/test_refimpl_x_gts_final_abstract.py]: Fixed — added `expected_id` optional parameter to `_validate_instance` and applied it to all four failure-path calls (abstract reject, abstract chain, abstract anon, abstract+final lifecycle).

**Review by Artifizer — APPROVED** [review]

File: README.md
Comment (Artifizer) [inline README.md]: Let's update version to 0.9 and put notes

Comment (Artifizer) [inline README.md:109]: 0.9 ...

Comment (aviator5) [reply README.md:109]: added

Comment (aviator5) [reply README.md]: updated

**Review by Artifizer — APPROVED** [review]
